components/tidb_query_expr/src/impl_like.rs

```diff
@@ -66,6 +70,87 @@ pub fn like<C: Collator>(target: BytesRef, pattern: BytesRef, escape: &i64) -> R
     Ok(Some(true as i64))
 }
 
+#[rpn_fn(capture = [metadata], metadata_mapper = init_regexp_utf8_data)]
+#[inline]
+pub fn regexp_utf8(
+    metadata: &Option<Regex>,
+    target: BytesRef,
+    pattern: BytesRef,
+) -> Result<Option<i64>> {
+    let target = match String::from_utf8(target.to_vec()) {
+        Ok(target) => target,
+        Err(err) => return Err(box_err!("invalid input value: {:?}", err)),
+    };
+
+    match metadata {
+        Some(regex) => Ok(Some(regex.is_match(target.as_ref()) as i64)),
+        None => {
+            let pattern = match String::from_utf8(pattern.to_vec()) {
+                Ok(pattern) => pattern,
+                Err(err) => return Err(box_err!("invalid input value: {:?}", err)),
+            };
+            let pattern = format!("(?i){}", pattern);
+
+            match BytesRegex::new(&pattern) {
+                Ok(bytes_regex) => Ok(Some(bytes_regex.is_match(target.as_ref()) as i64)),
+                Err(err) => Err(box_err!("invalid regex pattern: {:?}", err)),
+            }
+        }
+    }
+}
+
+#[rpn_fn(capture = [metadata], metadata_mapper = init_regexp_data)]
+#[inline]
+pub fn regexp(
+    metadata: &Option<BytesRegex>,
+    target: BytesRef,
+    pattern: BytesRef,
+) -> Result<Option<i64>> {
+    match metadata {
+        Some(regex) => Ok(Some(regex.is_match(target.as_ref()) as i64)),
+        None => {
+            let pattern = match String::from_utf8(pattern.to_vec()) {
+                Ok(pattern) => pattern,
+                Err(err) => return Err(box_err!("invalid input value: {:?}", err)),
+            };
+            match BytesRegex::new(&pattern) {
+                Ok(bytes_regex) => Ok(Some(bytes_regex.is_match(target.as_ref()) as i64)),
+                Err(err) => Err(box_err!("invalid regex pattern: {:?}", err)),
+            }
+        }
+    }
+}
+
+fn init_regexp_utf8_data(expr: &mut Expr) -> Result<Option<Regex>> {
+    let children = expr.mut_children();
+    let n = children.len();
+    assert!(n > 1);
+
+    let tree_node = &mut children[1];
+    match tree_node.get_tp() {
+        ExprType::ScalarFunc | ExprType::ColumnRef | ExprType::Null => Ok(None),
+        _ => match Regex::new(String::from_utf8(tree_node.take_val()).unwrap().as_ref()) {
+            Ok(regex) => Ok(Some(regex)),
+            Err(_) => Ok(None),
+        },
+    }
+}
+
+fn init_regexp_data(expr: &mut Expr) -> Result<Option<BytesRegex>> {
+    let children = expr.mut_children();
+    let n = children.len();
+    assert!(n > 1);
+
+    let tree_node = &mut children[1];
+    match tree_node.get_tp() {
+        ExprType::ScalarFunc | ExprType::ColumnRef | ExprType::Null => Ok(None),
+        _ => match BytesRegex::new(String::from_utf8(tree_node.take_val()).unwrap().as_ref()) {
```

Comment:
Would be better to use `?` instead of unwrap

Reply:
It is tried, while ` the trait ``std::convert::From<std::string::FromUtf8Error>`` is not implemented for ``tidb_query_common::error::EvaluateError`` ` will be reported.

Maybe we can implement it?